common/djangoapps/student/views.py

```diff
@@ -587,7 +589,11 @@ def dashboard(request):
     # Get any programs associated with courses being displayed.
     # This is passed along in the template context to allow rendering of
     # program-related information on the dashboard.
+    credential_configuration = CredentialsApiConfig.current()
     course_programs = _get_course_programs(user, [enrollment.course_id for enrollment in course_enrollments])
+    program_credentials = []
+    if credential_configuration.enabled:
```

Comment:
Look at where this check is done when retrieving programs. It's inside the helper function used to retrieve the data. You only need to add a single line here if you implement credential retrieval similarly.